src/ReactionNetwork.jl

```diff
@@ -0,0 +1,343 @@
+"""
+Macro that inputs an expression corresponding to a reaction network and output a Reaction Network Structure that can be used as input to generation of SDE and ODE and Jump problems.
+Most arrows accepted (both right, left and bi drectional arrows).
+Note that while --> is a correct arrow, neither <-- nor <--> works.
+Using non-filled arrows (⇐, ⟽, ⇒, ⟾, ⇔, ⟺) will disable mass kinetics and lets you cutomize reaction rates yourself.
+Use 0 or ∅ for degradation/creation to/from nothing.
+Example systems:
+    rn = @reaction_network_new rType begin #Creates a reaction network of type rType.
+        2.0, X + Y --> XY                  #This will have reaction rate corresponding to 2.0*[X][Y]
+        2.0, XY ← X + Y                    #Identical to 2.0, X + Y --> XY
+        (2.0,1.0), X + Y ↔ XY              #Identical to reactions (2.0, X + Y --> XY) and (1.0, XY --> X + Y).
+        2.0, X + Y ⟾ XY                   #Ignores mass kinetics. This will have reaction rate corresponding to 2.0*[X].
```

Comment:
There is a minor ambiguity in the meaning of the `⟾` arrow. The example suggests the reaction's total rate depends only on the first substrate. I think this is a reasonable convention, but should be made explicit to a user.

Otherwise, (as a contrived example) we may have to think about what `2.0, X + Y + Z ⟾ XYZ` ought to mean.

Reply:
Ok, this is actually a problem in my documentation (I think I changed the expression at some point to make them as simple as possible, but forgot to change some of the text.

The reaction rate in ´2.0, X + Y ⟾ XY´ will become 2.0. Basically it will only take whatever you write as reaction rate and not make any modifications. You can still write the reaction as ´2.0*X, X + Y ⟾ XY´ if you want reaction rate ´2.0*X´ (that is actually another non obvious thing that I probably should add...). You can put pretty much anything in the reaction rate as long as it do not contain variables not passed as parameters or functions which you have defined yourself. E.g.
´´´julia
rn = @reaction_network_new type begin
        log(X,2.0), X + Y --> XY      
end 
´´´
should work fine (although I am unsure when you would want that). However
´´´julia
hill2(x,v,K) = v*x*x/(K*K+x*x)
rn = @reaction_network_new type begin
        hill2(X,3.0,1.5), X + Y --> XY      
end 
´´´
will not work. the hill function and m.m. function works because I have defined those internally in the package.

I will update it so that it actually says what ´⟾´ does, as well as what you can put as rates.